cytopipe/cli/cmd.py

```diff
@@ -27,59 +27,76 @@ def run_cmd() -> None:
     -------
     None
     """
-    # checking provided cli arguments
-    cli_check(args_list=sys.argv)
+    # create args handler
+    # -- Cli Control Panel
+    args_handler = CliControlPanel(sys.argv)
+    # print(args_handler.__repr__())
 
-    # collecting parameters
-    params = sys.argv[1:]
-    mode_type = params[0]
-
-    # parsing arguments based on modes
-    if mode_type == "init":
+    # checking is user wanted to cli help
+    if args_handler.cli_help is True:
+        print(cli_docs)
+        sys.exit(0)
 
-        # parsing inputs
-        init_args = parse_init_args(params)
+    match args_handler.mode:
 
-        # setting up file paths
-        barcode_path = str(Path(init_args.barcode).absolute())
-        metadata_path = str(Path(init_args.metadata).absolute())
+        case "init":
 
-        # create data folder
-        # raises error if data directory exists (prevents overwriting)
-        data_dir_obj = Path("data")
-        data_dir_obj.mkdir(exist_ok=False)
-        data_dir_path = str(data_dir_obj.absolute())
+            # if mode help flag is shown, show cli help doc
+            if args_handler.mode_help is True:
+                print(init_doc)
+                sys.exit(0)
 
-        # moving all input files
-        for data_file in init_args.data:
-            f_path = str(Path(data_file).absolute())
-            shutil.move(f_path, data_dir_path)
+            # setting up input files for cytopipe
+            print("INFO: Formatting input files")
+            init_args = args_handler.parse_init_args()
 
-        shutil.move(barcode_path, data_dir_path)
-        shutil.move(metadata_path, data_dir_path)
+            # setting up file paths
+            barcode_path = str(Path(init_args.barcode).absolute())
+            metadata_path = str(Path(init_args.metadata).absolute())
 
-    elif mode_type == "run":
+            # create data folder
+            # raises error if data directory exists (prevents overwriting)
+            data_dir_obj = Path("data")
+            data_dir_obj.mkdir(exist_ok=False)
```

Comment:
What if force == True?

Reply:
I do not know if I explained to you about `snakemake's` ability to track the progress of the workflow and sets up save points for each successful rule. This is beautifully illustrated from this [paper](https://www.nature.com/articles/s41592-021-01254-9) on figure 1 far right figure. 

if `--force` where to be false, this indicates that when you re-run `cytopipe` it will start from where it failed. If `--force` where to be `True`, then executing `cytopipe run` again will cause the workflow to start from the beginning and recreate the files again. 

Hopefully that clears any confusion!

Reply:
ooohh, so if the directory is created during cytopipe, if the cytopipe run fails, the directory will be deleted?

(i can't see fig 1 b/c paywall)

Reply:
Not necessarily. Remember, the `data` directory is important to maintain. That's how `snakemake` knows where to put the outputs files. What occurs is that `snakemake` knows that there are files within the `data`  folder and when `--force` flag is `True`, you are telling `snakemake` (via `cytopipe`) "Hey, I know there are files there in there, but I want you to start from the beginning and overwrite them" 

The data directory does not get deleted.  The contents within the `data` folder gets overwritten. 

Hopefully this makes sense!

P.S: I have sent the paper I was referring via Discord!